### DISCUSSION TITLE
Time limit for provisioning

### DISCUSSION THREAD

**Discussion by H4usi** [schorschii/OCO-Server#49] [Ideas]
Hi,

This idea applies to both the server and the agent. It would be very helpful if we could set a time limit for installations or packages.

For example, it’s unlikely that installing 7zip would take more than 10 minutes. If it does, something has probably gone wrong. The problem is that the agent currently waits indefinitely for the installation to finish.

This idea didn’t come out of nowhere: Due to a typo, I accidentally created an infinite loop that wouldn’t stop—just like a “good” loop should 😅.

I had to restart the agent service to make it responsive again.

Have a great day
Andre

**Comment by schorschii** [comment]
Sounds like a useful feature. However, since I test new packages on a separate machine before general rollout, I currently don't see the necessity to implement it now. But if you need it for your environment, you may hire me to implement it. If you are interested you can [contact me](https://georg-sieber.de/?page=impressum) for a quote.

**Reply by H4usi** [reply]
We think about it. We are still in the introductory phase and are gathering ideas and experience. I'll come back to you when everything has settled down a bit. 

And yes, perhaps we should test it a little more extensively before we roll out a package 😊
